code/game/objects/items/twohanded.dm

```diff
@@ -500,6 +500,18 @@
 	actions_types = list(/datum/action/item_action/startchainsaw)
 	var/on = FALSE
 
+/obj/item/twohanded/required/chainsaw/suicide_act(mob/living/carbon/user)
+	if(on)
+		user.visible_message("<span class='suicide'>[user] begins to tear [user.p_their()] head off with [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(src, "sound/weapons/chainsawhit.ogg", 100, 1)
+		var/obj/item/bodypart/head/myhead = user.get_bodypart("head")
+		if(myhead)
+			myhead.dismember()
+	if(!on) //this may look like shitcode but i had to, for some reason it wouldn't recognize the if(on) above
```

Comment:
an `else` does not work?

Reply:
yeah, it doesn't recognize the if I promise. I have NO CLUE WHY.

Reply:
What do you mean "doesn't recognize the If"

Reply:
the one way above, or something like that.

Reply:
No, it will work, you're just doing it wrong.